MycroftAI/mycroft-core
# Feature/boot performance

## Description
Half of the time it takes to boot the device is consumed by interactions with MSM.  Skill-related data just doesn't change that frequently.  Use caching of MSM data to speed cut boot time in half. There is a related PR for MSM that has the same branch name.

## How to test
At its core, this is a refactor.  The logic hasn't changed, how frequently it executes has.  Boot up a device and ensure all still works as expected.  Install, remove and update a skill to ensure those actions reset the cached values.

## Contributor license agreement signed?
CLA [yes]
